src/Microsoft.AspNet.OData.Shared/Query/Expressions/FilterBinder.cs

```diff
@@ -399,7 +400,7 @@ public virtual Expression BindCollectionConstantNode(CollectionConstantNode node
             // so using a foreach loop and doing an implicit cast from object to the CLR type of ItemType.
             foreach (ConstantNode item in node.Collection)
             {
-                castedList.Add(item.Value);
+                castedList.Add(EnumDeserializationHelpers.ConvertEnumValue(item.Value, constantType));
```

Comment:
For readability, I would prefer to break this into two lines; one that tries to convert the enum value and the second that does the add.